crates/uv-resolver/src/requires_python.rs

```diff
@@ -70,37 +69,39 @@ impl RequiresPython {
     pub fn intersection<'a>(
         specifiers: impl Iterator<Item = &'a VersionSpecifiers>,
     ) -> Result<Option<Self>, RequiresPythonError> {
-        // Convert to PubGrub range and perform an intersection.
-        let range = specifiers
-            .into_iter()
-            .map(crate::pubgrub::PubGrubSpecifier::from_release_specifiers)
-            .fold_ok(None, |range: Option<Range<Version>>, requires_python| {
-                if let Some(range) = range {
-                    Some(range.intersection(&requires_python.into()))
-                } else {
-                    Some(requires_python.into())
+        let mut combined: BTreeSet<VersionSpecifier> = BTreeSet::new();
+        let mut lower_bound: LowerBound = LowerBound(Bound::Unbounded);
+        let mut upper_bound: UpperBound = UpperBound(Bound::Unbounded);
+
+        for specifier in specifiers {
+            // Convert to PubGrub range and perform an intersection.
+            let requires_python =
+                crate::pubgrub::PubGrubSpecifier::from_release_specifiers(specifier)?;
+            if let Some((lower, upper)) = requires_python.bounding_range() {
+                let lower = LowerBound(lower.cloned());
+                let upper = UpperBound(upper.cloned());
+                if lower > lower_bound {
+                    lower_bound = lower;
                 }
-            })?;
+                if upper < upper_bound {
+                    upper_bound = upper;
+                }
+            }
 
-        let Some(range) = range else {
-            return Ok(None);
-        };
+            // Track all specifiers for the final result.
+            combined.extend(specifier.iter().cloned());
+        }
 
-        // Extract the bounds.
-        let (lower_bound, upper_bound) = range
-            .bounding_range()
-            .map(|(lower_bound, upper_bound)| (lower_bound.cloned(), upper_bound.cloned()))
-            .unwrap_or((Bound::Unbounded, Bound::Unbounded));
+        if combined.is_empty() {
+            return Ok(None);
+        }
 
-        // Convert back to PEP 440 specifiers.
-        let specifiers = range
-            .iter()
-            .flat_map(VersionSpecifier::from_release_only_bounds)
-            .collect();
```

Comment:
This isn't quite correct -- we can't just AND these, since a `!=` specifier gets converted to `<` and `>` the value. So instead we retain the actual specifiers now.